068-performance-testing.md

```diff
@@ -0,0 +1,451 @@
+
+- This method integrates performance testing directly alongside existing system tests. 
+- It leverages the already established infrastructure and testing utilities, ensuring that performance and system tests can share resources such as the ResourceManager, LogCollector, and MetricsCollector.
+- **Advantages**:
+  - Streamlines the development process by using existing test infrastructure.
+  - Facilitates immediate feedback on performance impacts within the same testing framework used for system tests.
+  - Reduces the overhead of maintaining separate modules for system and performance tests.
+
+### Core parts
+
+The following strategies will be core to our implementation:
+
+1. **JUnit5 Usage:** 
+   1. Employ JUnit5 for structuring and executing performance tests. This choice allows us to benefit from JUnit5's advanced features such as dynamic tests, parameterized tests, and its powerful extension model, enhancing our test design and flexibility.
+2. **Resource Management:** 
+   1. Utilize the ResourceManager from the system test module to manage the lifecycle of resources.
+   2. This approach ensures efficient setup, teardown, and management of Kafka clusters, topics, and other necessary components across both performance and system tests.
+3. **Metrics Collection:** 
+   1. Base our metrics collection on the system test module's MetricsCollector, with plans to possibly extend this functionality in the future from an external library like Test Frame. 
+   2. Additionally, introduce specialized collectors such as TopicOperatorMetricsCollector to gather detailed metrics relevant to the Strimzi Topic Operator, tailoring our data gathering to the nuances of Kafka and Strimzi.
+4. **Performance Reporting:** 
+   1. Develop a performance test reporting mechanism. This component will be essential for documenting and analyzing the outcomes of performance tests. 
+   2. The reporter should be capable of generating detailed summaries and insights, 
+   ideally supporting formats conducive to both human reading and automated parsing, such as Markdown or structured JSON.
+
+**Including within the `systemtest` module:** Integrate performance tests to ensure they coexist seamlessly with system tests, potentially categorizing tests to maintain clarity and prevent any confusion between system and performance testing efforts (i.e., using performance profile). 
+
+### CI: Azure Pipelines
+
+We assume Azure machines may not suffice for all performance tests, meaning we anticipate that lower-complexity performance tests, 
+not requiring as much memory/CPU, could run on such infrastructure. However, for tests needing more memory in the context of 
+performance, we prefer another approach (i.e., using a testing farm) to execute these tests.
+
+### CI: Possible integration with Testing farm 
+
+Special consideration will be given to the potential integration of performance tests with testing farms, utilizing 
```

Comment:
the `testing farms` here seems weird
if you are writing about Testing Farm, then it should have the name in the text
otherwise, if you think of something else, you should name it here I think

Reply:
Testing farm fixed